src/PayID/pay-id-components.ts

```diff
@@ -6,16 +6,19 @@
  * A class which encapsulates components of a Pay ID.
  */
 export default class PayIdComponents {
+  public readonly host: string
+  public readonly path: string
+
   /**
    * Create a new PayIdComponents.
    *
    * @param host - The host of the payment pointer.
    * @param path - The path of the payment pointer, starting with a leading '/'.
    */
-  public constructor(
-    public readonly host: string,
-    public readonly path: string,
-  ) {}
+  public constructor(host: string, path: string) {
+    this.host = host
+    this.path = path
```

Comment:
There's [a rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-parameter-properties.md) that disallows declaring class properties in the constructor. I'm ok with disabling this rule for now, but since _some_ properties might not be declarable in the constructor, it makes sense to be consistent.

Reply:
I'm happy to use lint rules if that's what folks like doing.

Reply:
The only reason I could think of not liking this style is that you forget to assign a variable. However, this appears to not be the case, since the following errors:

```
public constructor(host: string, path: string) {
    this.host = host
}
```

Yells taht I didn't initialize path.